core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -127,6 +128,16 @@
     public static final Setting<Integer> MAX_SLICES_PER_SCROLL = Setting.intSetting("index.max_slices_per_scroll",
         1024, 1, Property.Dynamic, Property.IndexScope);
 
+    /**
+     * The number of shard copies required to be active before returning from an index creation operation.
+     */
+    public static final Setting<ActiveShardCount> WAIT_FOR_ACTIVE_SHARDS_SETTING =
```

Comment:
well, taking this a step further, the setting idea was to be used as a replacement for the current action setting that is a default for the indexing (i.e. replication) operations. I think that's a good idea as well as making the setting an index setting. I was thinking that if we have that setting, do we really need a request level parameter for the index creation or can we just fall back to the setting (especially if they are dynamic). Thinking about it more, I think request parameters OK. I'm not sure about `index. wait_for_active_shards` . Can't come up with a better one though. Will think about it more..